### PR TITLE
8293011: riscv: Duplicated stubs to interpreter for static calls

### PR DESCRIPTION
**Pull request by DingliZhang**
Follow up [JDK-8280481](https://bugs.openjdk.org/browse/JDK-8280481).
Calls of Java methods have stubs to the interpreter for the cases when an invoked Java method is not compiled. Calls of static Java methods and final Java methods have statically bound information about a callee during compilation. C1 and C2 always generate a new stub for each call. As the generated stubs for calls of the same method are the same, they can be shared.

## Testing:

- hotspot/jdk tier1 on unmatched board
- hotspot/jtreg/compiler/sharedstubs/SharedStubToInterpTest.java on qemu

#### Results from [Renaissance 0.14.0](https://github.com/renaissance-benchmarks/renaissance/releases/tag/v0.14.0)
Note: 'Nmethods with shared stubs' is the total number of nmethods counted during benchmark's run. 'Final # of nmethods' is a number of nmethods in CodeCache when JVM exited.

- riscv64
```
+------------------+-------------+----------------------------+---------------------+
| Benchmark        | Saved bytes | Nmethods with shared stubs | Final # of nmethods |
+------------------+-------------+----------------------------+---------------------+
| dotty            |     1099488 |                       4483 |               12447 |
| dec-tree         |      511296 |                       2310 |               18583 |
| naive-bayes      |      358128 |                       1714 |                8677 |
| log-regression   |      365136 |                       1662 |               14626 |
| als              |      444576 |                       2107 |                8464 |
| finagle-chirper  |      265584 |                       1558 |               11003 |
| movie-lens       |      397776 |                       2106 |                6842 |
| finagle-http     |      160656 |                        974 |                7243 |
| page-rank        |      246672 |                       1261 |               10293 |
| chi-square       |      196080 |                        992 |                8841 |
| akka-uct         |      138672 |                        595 |                4564 |
| reactors         |       57552 |                        328 |                2338 |
| scala-stm-bench7 |       42122 |                        254 |                2261 |
| philosophers     |       45744 |                        241 |                1945 |
| scala-doku       |       48624 |                        213 |                 794 |
| rx-scrabble      |       46128 |                        271 |                1945 |
| future-genetic   |       37248 |                        234 |                1818 |
| scrabble         |       30384 |                        170 |                1628 |
| par-mnemonics    |       28176 |                        137 |                1317 |
+------------------+-------------+----------------------------+---------------------+
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293011](https://bugs.openjdk.org/browse/JDK-8293011): riscv: Duplicated stubs to interpreter for static calls

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10057/head:pull/10057` \
`$ git checkout pull/10057`

Update a local copy of the PR: \
`$ git checkout pull/10057` \
`$ git pull https://git.openjdk.org/jdk pull/10057/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10057`

View PR using the GUI difftool: \
`$ git pr show -t 10057`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10057.diff">https://git.openjdk.org/jdk/pull/10057.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/riscv/codeBuffer_riscv.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```
Comment (feilongjiang) [inline src/hotspot/cpu/riscv/codeBuffer_riscv.cpp:2]: Missing copyright of your company for the new file.

**Review by feilongjiang — CHANGES_REQUESTED** [review]
Changes look good, except for one copyright issue.

**Comment by DingliZhang** [comment]
> Changes look good, except for one copyright issue.

Thank you, @feilongjiang!  I have added the missing copyright.

**Review by RealFYang — APPROVED** [review]
Looks fine.

**Comment by DingliZhang** [comment]
@RealFYang @feilongjiang Thank you for the reviews!
/integrate

**Pull request closed without merging**